mbuild/formats/lammpsdata.py

```diff
@@ -289,6 +260,43 @@ def write_lammpsdata(
     if impropers and imp_dihedrals:
         raise ValueError("Use of multiple improper styles is not supported")
 
+    # Get lj conversion factors if they exist and apply lj params to charges and box
+    # Params are in read in mbuild units of angstrom, kcal/mol, amu and converted to lammps units
+    if unit_style == "lj":
+        sigma_conversion_factor = _evaluate_lj_conversion_factors(
+            structure, "sigma", sigma_conversion_factor
+        )
+        epsilon_conversion_factor = _evaluate_lj_conversion_factors(
+            structure, "epsilon", epsilon_conversion_factor
+        )
+        mass_conversion_factor = _evaluate_lj_conversion_factors(
+            structure, "mass", mass_conversion_factor
+        )
+        # Convert coordinates and charges to LJ units
+        xyz = xyz / sigma_conversion_factor
+        charges = (charges * ELEM_TO_COUL) / np.sqrt(
```

Comment:
Can you explain why the conversion from e to Coulomb is needed? I agree with the denominator but I am unsure about the numerator. See link below to how HOOMD handles charge conversions:

https://hoomd-blue.readthedocs.io/en/stable/units.html

Also, why is 10^-6 needed?

Reply:
In order to remove dimensions, the dimensions of every component needs to be tabulated and converted to match.

From the link you posted from hoomd: 

```math
charge = charge (elem charge) / (4*pi*epsilon_0*sigma*epsilon)**(0.5)
```
So we need to make sure the units all match up from this equation.
To do so:
Charge --> convert to coulombs using ELEM_TO_COUL
epsilon_0 --> Given in Farads/meter, which is equivalent to (s^2 * C^2 * kg^-1 * m^-3)
epsilon --> Given in kCal, so convert to kJ using KCAL_TO_KJ. This then can be broken into further SI units: 
(1000 * kg * m^2 * s^-2)
sigma --> given in angstrom. To convert to m, multiply by 10^-10

Put it all together:
charge * ELEM_TO_COUL / sqrt(
(4 * pi * epsilon_0) * (epsilon * KCAL_TO_KJ * 1000) *  (sigma * 10^-10)

This can be equivalently broken into how it is shown in the code:
charge * ELEM_TO_COUL / sqrt(
(4 * pi * epsilon_0) * (epsilon * KCAL_TO_KJ * 1000) *  (sigma * 10^-9 * 10^-1)
charge * ELEM_TO_COUL / sqrt(
(4 * pi * epsilon_0) * (epsilon * KCAL_TO_KJ * 1000) *  (sigma * 10^-9 * NM_TO_ANG^-1)
This was just done to get the sigma and epsilon parameters into SI units. But the 10^-6 comes from the 10^-9 and the 1000.

To double check the units, we see:
C / sqrt((s^2 * C^2 * kg^-1 * m^-3) * (kg * m^2 * s^-2) * (m))        :::::: remove seconds squared
= C /  sqrt((C^2 * kg^-1 * m^-3) * (kg * m^2) * (m))   :::::::::: remove kg
= C /  sqrt((C^2 * m^-3) * (m^2) * (m))   ::::::::::::::: remove meters cubed
= C /  sqrt((C^2)) :::::::::::: remove coulombs
= C/C
= 1   

I can write a unit test for this though, you're right that should definitely be there.

Reply:
Makes sense, thanks!!